ctapipe/image/extractor.py

```diff
@@ -209,15 +214,38 @@ def subtract_baseline(waveforms, baseline_start, baseline_end):
     baseline_corrected : ndarray
         Waveform with the baseline subtracted
     """
-    baseline_corrected = waveforms - np.mean(
-        waveforms[..., baseline_start:baseline_end], axis=-1
-    )[..., None]
+    baseline_corrected = (
+        waveforms
+        - np.mean(waveforms[..., baseline_start:baseline_end], axis=-1)[..., None]
+    )
 
     return baseline_corrected
 
 
-class ImageExtractor(Component):
+def slide_window(waveform, window_width):
+    """Smooth a pixel's waveform with a kernel of certain size via convolution.
+
+    Parameters
+    ----------
+    waveforms : array_like
+        DL0-level waveforms of one event.
+        Shape: (n_samples)
+    window_width : int
+        Size of the smoothing kernel.
+
+    Returns
+    -------
+    sum : array_like
+        Array containing the sums for each of the kernel positions.
+        Shape: (n_samples - window_width - 1)
+
+    Dr. Michele Peresano, 2019
```

Comment:
I'd suggest removing your name here, and instead adding it to the `CODEOWERS` file at the top level (we decided against putting names in the comments).  You also should also update `AUTHORS` and `.mailmap` with your information

Reply:
However, if there is a reference for the MARS analysis, it would be nice to add it also to `docs/references.rst` and add a citation link.  Note that it should not be in the `Returns` section (sphinx won't like that), but in either the description or `See Also` section (see https://numpydoc.readthedocs.io/)

Reply:
I'll try to see what could be the right citation

Reply:
If I understood it correctly is this the two-pass integration method described in section 4 of https://ui.adsabs.harvard.edu/abs/2006APh....25..391H/abstract ? 

If this is correct, then it maybe shouldn't be called CTA/mars analysis :-)

Reply:
Hi Gernot, I apologize for the late reply.

I don't know if the version of the algorithm I coded here comes directly from the reference that you provide (in any case, from what you say it seems a necessary reference to be added to the documentation so I will do it).
I just tried with as much fidelity as possible to translate in _ctapipe_-language the version that is used inside CTA/MARS.
That is the reason for the name of the PR (if this is what you refer to).
I also explicited in the class documentation that `"[...] This is in particular the CTA-MARS version."`.

For me there is no problem in changing names, perhaps we can ask @moralejo for a confirmation on this.

Reply:
Hi, the Eventdisplay reference is indeed to be included:  I implemented this in MARS for the CTA analysis (it is not used in MAGIC) to get closer to Eventdisplay. Still, in the spirit of having fully independent analysis codes, I just took the idea, "make a time fit with significant pixels, then do a second round of pulse integration on the rest of the pixels using the times 'predicted' by the fit". The implementation was done in a completely independent manner, and it is bound to differ in the details. I do not think it makes much sense to copy every detail of the Mars version for protopipe - as you @HealthyPear realized, some things (like ROOT's "robust" fit details) would take quite some effort and are probably not essential.

Reply:
> Hi, the Eventdisplay reference is indeed to be included: I implemented this in MARS for the CTA analysis (it is not used in MAGIC) to get closer to Eventdisplay. Still, in the spirit of having fully independent analysis codes, I just took the idea, "make a time fit with significant pixels, then do a second round of pulse integration on the rest of the pixels using the times 'predicted' by the fit". The implementation was done in a completely independent manner, and it is bound to differ in the details. 

I will then add the Eventdisplay one; do you have any suggestion for the reference on the CTA/MARS side? Otherwise as I stated above I specified that this is the version used there.

Reply:
> I'd suggest removing your name here, 

Done

> and instead adding it to the `CODEOWERS` file at the top level (we decided against putting names in the comments). 

Done - I hope is the correct syntax

> You also should also update `AUTHORS` and `.mailmap` with your information

I have already done this sometime ago, do I have to change or update something?

Reply:
> 
> I will then add the Eventdisplay one; do you have any suggestion for the reference on the CTA/MARS side? Meanwhile, I took the liberty to add the URL of the wiki page as a citation.

Unfortunately we have no public reference for that, since as I said it is not a method we use in the MAGIC analysis. Wiki page is ok for internal reference.

Reply:
possibly the ICRC proceeding by Maxim Shayduk - I think that's the origin of the method.